### PR TITLE
Add Boolean Prototype Dispatcher

### PR DESCRIPTION
**Pull request by Yuyupo**
JerryScript-DCO-1.0-Signed-off-by: Daniella Barsony bella@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/ecma/builtin-objects/ecma-builtin-boolean-prototype.c [hunk 1]
```diff
@@ -105,6 +91,55 @@ ecma_builtin_boolean_prototype_object_value_of (ecma_value_t this_arg) /**< this
...
+
+      JERRY_ASSERT (ecma_is_value_boolean (value_of_ret));
+
+      return ecma_make_magic_string_value (LIT_MAGIC_STRING_FALSE);
+    }
+    case ECMA_BOOLEAN_PROTOTYPE_ROUTINE_VALUE_OF:
+    {
+      return ecma_builtin_boolean_prototype_object_value_of (this_arg);
+    }
+
```
Comment (dbatyai) [inline jerry-core/ecma/builtin-objects/ecma-builtin-boolean-prototype.c:135]: Please remove this newline.

**Review by dbatyai — CHANGES_REQUESTED** [review]

File: jerry-core/ecma/builtin-objects/ecma-builtin-boolean-prototype.c [hunk 2]
```diff
@@ -105,6 +91,54 @@ ecma_builtin_boolean_prototype_object_value_of (ecma_value_t this_arg) /**< this
...
+                                                                               *   identifier */
+                                                 ecma_value_t this_arg, /**< 'this' argument value */
+                                                 const ecma_value_t arguments_list_p[], /**< list of arguments
+                                                                                         *   passed to routine */
+                                                 uint32_t arguments_number) /**< length of arguments' list */
+{
+  JERRY_UNUSED (arguments_number);
+  JERRY_UNUSED (arguments_list_p);
+
+  switch (builtin_routine_id)
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-boolean-prototype.c:111]:
```c
ecma_value_t value_of_ret = ecma_builtin_boolean_prototype_object_value_of (this_arg);

if (id == ECMA_BOOLEAN_PROTOTYPE_ROUTINE_VALUE_OF) return value_of_ret;

perform toString operation.
```

So no switch is needed.

**Review by rerobika — CHANGES_REQUESTED** [review]

**Review by dbatyai — APPROVED** [review]
LGTM

File: jerry-core/ecma/builtin-objects/ecma-builtin-boolean-prototype.c [hunk 3]
```diff
@@ -105,6 +91,44 @@ ecma_builtin_boolean_prototype_object_value_of (ecma_value_t this_arg) /**< this
...
+  JERRY_UNUSED (arguments_number);
+  JERRY_UNUSED (arguments_list_p);
+
+  ecma_value_t value_of_ret = ecma_builtin_boolean_prototype_object_value_of (this_arg);
+  if (builtin_routine_id == ECMA_BOOLEAN_PROTOTYPE_ROUTINE_VALUE_OF)
+  {
+    return value_of_ret;
+  }
+
+  if (ECMA_IS_VALUE_ERROR (value_of_ret))
```
Comment (zherczeg) [inline jerry-core/ecma/builtin-objects/ecma-builtin-boolean-prototype.c:117]: There should be an assert before this if.
`JERRY_ASSERT (builtin_routine_id == ECMA_BOOLEAN_PROTOTYPE_ROUTINE_TO_STRING);`

**Review by zherczeg — CHANGES_REQUESTED** [review]

**Review by rerobika — APPROVED** [review]
LGTM

File: jerry-core/ecma/builtin-objects/ecma-builtin-boolean-prototype.c [hunk 4]
```diff
@@ -105,6 +91,46 @@ ecma_builtin_boolean_prototype_object_value_of (ecma_value_t this_arg) /**< this
...
+  {
+    return value_of_ret;
+  }
+
+  if (ecma_is_value_true (value_of_ret))
+  {
+    return ecma_make_magic_string_value (LIT_MAGIC_STRING_TRUE);
+  }
+
+  JERRY_ASSERT (ecma_is_value_boolean (value_of_ret));
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-boolean-prototype.c:129]: `ecma_is_value_boolean` -> `ecma_is_value_false`

**Review by rerobika — CHANGES_REQUESTED** [review]

**Review by rerobika — APPROVED** [review]
LGTM

**Review by zherczeg — APPROVED** [review]
LGTM

**Pull request merged**